ansible_events_ui/api/activation.py

```diff
@@ -1,8 +1,49 @@
 """Activation API endpoints."""
-from fastapi import APIRouter
+import logging
+
+import sqlalchemy as sa
+from fastapi import APIRouter, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui import schemas
+from ansible_events_ui.db import models
+from ansible_events_ui.db.dependency import get_db_session
+
+logger = logging.getLogger("ansible_events_ui")
 
 __all__ = ("router",)
 
 router = APIRouter()
 
-# Implement activation API here.
+
+class ActivationQueryError(Exception):
+    def __init__(self, err_msg):
+        self.message = err_msg
+        super().__init__(self.message)
+
+
+@router.post("/api/activation/")
+async def create_activation(
+    activation: schemas.Activation,
+    db: AsyncSession = Depends(get_db_session),
+):
+    try:
+        query = sa.insert(models.activations).values(
+            name=activation.name,
+            description=activation.description,
+            rulebook_id=activation.rulebook_id,
+            inventory_id=activation.inventory_id,
+            execution_env_id=activation.execution_env_id,
+            restart_policy_id=activation.restart_policy_id,
+            playbook_id=activation.playbook_id,
+            activation_enabled=activation.activation_enabled,
+            extra_var_id=activation.extra_var_id,
+        )
+        result = await db.execute(query)
+        await db.commit()
+        (id_,) = result.inserted_primary_key
+
+        return {**activation.dict(), "id": id_}
+    except Exception as e:
```

Comment:
1. Normally you should avoid using base class `Excetion` when catching exceptions, unless it's a top-level exception handler.
2. Exception handler should except specific exceptions you expect to occur. In this case specific case it's `sqlalchemy.exc.IntegrityError`.
3. `try` clause should be narrowed to the scope where an exception is expected. In this case `db.execute` will raise this exception, so only this call should be in the `try` clause.
4. Custom exception doesn't change the resulting behavior. You should raise `HTTPException` with code `422`.
5. Please add test case to test this behavior.